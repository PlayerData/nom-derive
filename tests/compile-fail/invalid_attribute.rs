extern crate nom;
extern crate nom_derive;
use nom_derive::Nom;

#[derive(Nom)]
pub struct S {
    #[nom(Parse=u32)]
    a: u32,
}

#[derive(Nom)]
#[nom(Parse="be_u32")] // ERROR: Attribute Parse(be_u32) is not valid for top-level
pub struct S2 {
    a: u32,
}

#[derive(Nom)]
pub struct S3 {
    #[nom(Exact)] // ERROR: Attribute Exact is not valid for field-level
    a: u32,
}

fn main() {}