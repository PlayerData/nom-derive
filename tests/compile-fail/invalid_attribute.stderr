error: expected literal
 --> $DIR/invalid_attribute.rs:7:17
  |
7 |     #[nom(Parse=u32)]
  |                 ^^^

error: Attribute Parse(be_u32) is not valid for top-level
  --> $DIR/invalid_attribute.rs:11:10
   |
11 | #[derive(Nom)]
   |          ^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: Attribute Exact is not valid for field-level
  --> $DIR/invalid_attribute.rs:17:10
   |
17 | #[derive(Nom)]
   |          ^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
